### PR TITLE
Move to core 22 (5.0-edge)

### PR DESCRIPTION
**Pull request by mihalicyn**
- move to core22
- fix ZFS 0.6/0.7 build issues on core22

### CODE REVIEW

**Comment by mihalicyn** [comment]
ZFS 0.6:
https://github.com/mihalicyn/zfs/commits/zfs-0.6-core22-compilation-fix
 
ZFS 0.7:
https://github.com/mihalicyn/zfs/commits/zfs-0.7-core22-compilation-fix

I'm on the way to create `.patch` files and properly include them in the `lxd-pkg-snap`. But just for demonstration and testing purposes publishing this PR as it is.

File: snapcraft.yaml [hunk 1]
```diff
@@ -1054,6 +1049,12 @@ parts:
 
       git cherry-pick fb963d33ee0dd350143ba1c9cd35d5f7d86910d2  # Fix endian build problem on ppc64el
 
+      patch -p1 < "${SNAPCRAFT_PROJECT_DIR}/patches/zfs-0-6/0001-Add-support-for-libtirpc.patch"
+      patch -p1 < "${SNAPCRAFT_PROJECT_DIR}/patches/zfs-0-6/0002-config-better-libtirpc-detection.patch"
+      patch -p1 < "${SNAPCRAFT_PROJECT_DIR}/patches/zfs-0-6/0003-explicitly-define-____cacheline_aligned.patch"
+      patch -p1 < "${SNAPCRAFT_PROJECT_DIR}/patches/zfs-0-6/0004-Fix-static-data-to-link-with-fno-common.patch"
+      patch -p1 < "${SNAPCRAFT_PROJECT_DIR}/patches/zfs-0-6/0005-A-simplified-backport-of-2faf05612f9-libshare-smb-cl.patch"
```
Comment (simondeziel) [inline snapcraft.yaml:1010]: `s/SNAPCRAFT_PROJECT_DIR/CRAFT_PROJECT_DIR/` here too, please.

File: snapcraft.yaml [hunk 2]
```diff
@@ -1088,6 +1089,10 @@ parts:
         [ "$(uname -m)" != "s390x" ] && \
         [ "$(uname -m)" != "ppc64le" ] && exit 0
 
+      patch -p1 < "${SNAPCRAFT_PROJECT_DIR}/patches/zfs-0-7/0001-explicitly-define-____cacheline_aligned.patch"
+      patch -p1 < "${SNAPCRAFT_PROJECT_DIR}/patches/zfs-0-7/0002-Fix-static-data-to-link-with-fno-common.patch"
+      patch -p1 < "${SNAPCRAFT_PROJECT_DIR}/patches/zfs-0-7/0003-A-simplified-backport-of-2faf05612f9-libshare-smb-cl.patch"
```
Comment (simondeziel) [inline snapcraft.yaml:1047]: `s/SNAPCRAFT_PROJECT_DIR/CRAFT_PROJECT_DIR/` here too, please.

**Review by simondeziel — DISMISSED** [review]
LGTM, with the small inconsistency `s/SNAPCRAFT_PROJECT_DIR/CRAFT_PROJECT_DIR/` fixed.

That is a very welcome change, thank you!

File: snapcraft.yaml (hunk 1 above)
Comment (simondeziel) [reply snapcraft.yaml:1010]: Ah, I just realize you used the old variable name to be consistent with what was done in that part, please ignore my comment then. We'll change them all in a later PR.

File: snapcraft.yaml (hunk 2 above)
Comment (simondeziel) [reply snapcraft.yaml:1047]: Same here, please ignore.

**Review by MusicDin — DISMISSED** [review]

**Comment by tomponline** [comment]
LGTM, but wont merge until after LXD 5.0.3 is released.

**Comment by tomponline** [comment]
This can be rebased and we can then look to merge it soon. Ta

**Comment by simondeziel** [comment]
This would also be a good time to drop the `build-attributes` that no longer make sense if the base is `core22`, see https://snapcraft.io/docs/release-notes-snapcraft-6-1#heading--core22

**Comment by mihalicyn** [comment]
> This can be rebased and we can then look to merge it soon. Ta

Have rebased.

**Comment by mihalicyn** [comment]
> This would also be a good time to drop the `build-attributes` that no longer make sense if the base is `core22`, see https://snapcraft.io/docs/release-notes-snapcraft-6-1#heading--core22

Have done.

**Review by tomponline — APPROVED** [review]

**Pull request merged**